sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1552,7 +1553,8 @@ class Analyzer(override val catalogManager: CatalogManager)
 
     private def resolveMergeExprOrFail(e: Expression, p: LogicalPlan): Expression = {
       val resolved = resolveExpressionByPlanChildren(e, p)
-      resolved.references.filter(!_.resolved).foreach { a =>
+      resolved.references.filter { attribute: Attribute =>
+        !attribute.resolved && attribute.name != CURRENT_DEFAULT_COLUMN_NAME }.foreach { a =>
```

Comment:
What if there is an unresolved column "default" which is not default expression in the query?

Reply:
Good question. The `ResolveDefaultColumns` analysis rule exclusively handles this case. I added a comment here and a new unit test case.